nixos/modules/services/backup/burpServer.nix

```diff
@@ -0,0 +1,74 @@
+{ config, lib, networking, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.burpServer;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.burpServer = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable burp.";
+      };
+
+      configFile = mkOption {
+        default = "/etc/burp/burp.conf";
+        description = "Path to the burp config file. You need to provide a fully working configuration file and setup all paths yourself. burp will be start as user burp group burp and tcp port 4971 will be opened in the firewall. Ensure that you have setup the configuration accordingly.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.burp;
+        example = literalExample "pkgs.burp";
+        description = ''burp package to use.'';
+      };
+
+    };
+
+  };
+
+  ###### implementation
+
+  config = mkIf config.services.burpServer.enable {
+
+    users.extraUsers.burp = {
+      name = "burp";
+      group = "burp";
+      isSystemUser = true;
+      description = "burp server user";
+    };
+
+    users.extraGroups.burp = {
+      name = "burp";
+    };
+
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.services.burp = {
+      description = "burp server";
+
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/burp -F -c ${toString cfg.configFile}";
+        User = "burp";
+        Group = "burp";
+      };
+
+    };
+  
+    networking.firewall.allowedTCPPorts = [ 4971 ];
```

Comment:
I'd prefer if you mention that enabling this service also opens the requisite ports in the description, though you could argue that enabling a service implies also setting up everything required for it to function, including opening ports ...